### PR TITLE
LABS-1161 Fix the check for when to build delta and limit the size of delta

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -2401,8 +2405,12 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
         WT_ASSERT_ALWAYS(session, chunk->entries > 0, "Trying to write an empty chunk");
     }
 
-    if (last_block)
-        WT_RET(__rec_build_delta(session, r));
+    if (last_block) {
+        WT_RET(__rec_build_delta(session, r, &build_delta));
+        /* Discard the delta if it is larger than one tenth of the size of the full image. */
+        if (build_delta && r->delta.size > chunk->image.size / 10)
+            build_delta = false;
```
Comment (pmacko86) [inline src/reconcile/rec_write.c:2412]: I'm assuming the delta will get cleaned up at some later point if we're not using it?

File: src/reconcile/rec_write.c [hunk 2]
```diff
@@ -2401,8 +2405,12 @@ __rec_split_write(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK *chunk
         WT_ASSERT_ALWAYS(session, chunk->entries > 0, "Trying to write an empty chunk");
     }
 
-    if (last_block)
-        WT_RET(__rec_build_delta(session, r));
+    if (last_block) {
+        WT_RET(__rec_build_delta(session, r, &build_delta));
+        /* Discard the delta if it is larger than one tenth of the size of the full image. */
```
Comment (pmacko86) [inline src/reconcile/rec_write.c:2410]: Just curious, how was the 1/10 ratio determined?

**Review by pmacko86 — APPROVED** [review]
LGTM

File: src/reconcile/rec_write.c (hunk 1 above)
Comment (quchenhao) [reply src/reconcile/rec_write.c:2412]: yes, when the reconcile structure is destroyed.

File: src/reconcile/rec_write.c (hunk 2 above)
Comment (quchenhao) [reply src/reconcile/rec_write.c:2410]: I select this randomly for now. There is no data backing it. We can change or configure this in the future.

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**